RobustPGO/RobustSolver.cpp

```diff
@@ -133,33 +142,35 @@ void RobustSolver::addOdometry(const gtsam::NonlinearFactorGraph& odom_factor, c
   }
 }
 
-void RobustSolver::updateBatch(gtsam::NonlinearFactorGraph factors,
-    const gtsam::Values& values, const gtsam::Key& key0) {
+void RobustSolver::updateBatch(const gtsam::NonlinearFactorGraph& factors,
+                               const gtsam::Values& values, 
+                               const gtsam::Key& key0) {
 
   // load graph assumes that the previous graph has been cleared
   gtsam::Key current_key = key0; // initial key
   // note that as of now only deal with between factors)
   // first load the odometry
   // order a nonlinear factor graph as odometry first
   bool extracted_odom = false;
+  gtsam::NonlinearFactorGraph update_factors = factors; 
   while (!extracted_odom) {
     bool end_of_odom = true;
-    for (size_t i = 0; i < factors.size(); i++) {
+    for (size_t i = 0; i < update_factors.size(); i++) {
```

Comment:
We can make this a range-based for loop if we are using c++11:
```c++
for (gtsam::Pose3& factor : update_factors) {
```
I'm not actually sure if `gtsam::NonlinearFactorGraph` is iterable. If it is, probably the type needs to be the `gstam::Values` base and not `gtsam::Pose3`. Up to you @yunzc if this is worth doing.